app/uk/gov/hmrc/softdrinksindustrylevy/controllers/SdilController.scala

```diff
@@ -59,9 +62,33 @@ class SdilController @Inject()(desSubmissionService: DesSubmissionService,
   }
 
   def checkPendingSubscription(utr: String): Action[AnyContent] = Action.async { implicit request =>
-    buffer.find({"subscription.utr" -> utr}) map {
+    buffer.find("subscription.utr" -> utr) map {
       case Nil => NotFound(Json.obj("status" -> "SUBSCRIPTION_NOT_FOUND"))
       case _ => Ok(Json.obj("status" -> "SUBSCRIPTION_PENDING"))
     }
   }
+
+  def checkEnrolmentStatus(utr: String): Action[AnyContent] = Action.async { implicit request =>
+    desConnector.retrieveSubscriptionDetails("utr", utr) flatMap {
+      case Some(s) =>
+        buffer.find("subscription.utr" -> utr) flatMap {
+          case Nil => Future successful Ok(Json.toJson(s))
+          case l => taxEnrolmentConnector.getSubscription(l.head.formBundleNumber) flatMap {
```

Comment:
you can do `case l :: _` instead, and then you wouldn't have to do `l.head`

Reply:
Will change it.